### PR TITLE
Memory Limit added to RCConfig in autoscaling_utils.go

### PR DESCRIPTION
**Pull request by socaa**
Part of #11570 

@piosz

### CODE REVIEW

**Comment by socaa** [comment]
PTAL

@piosz

File: test/e2e/autoscaling_utils.go [hunk 1]
```diff
@@ -228,7 +230,12 @@ func runServiceAndRCForResourceConsumer(c *client.Client, ns, name string, repli
...
 		Timeout:   timeoutRC,
 		Replicas:  replicas,
+		CpuLimit:  int64(cpuLimit),     // millicores
+		MemLimit:  MBtoBytes(memLimit), // MemLimit is in bytes
 	}
 	expectNoError(RunRC(config))
+}
 
+func MBtoBytes(megabytes int) int64 {
+	return int64(megabytes * 1000000)
```
Comment (piosz) [inline test/e2e/autoscaling_utils.go]: 1024*1024

Comment (piosz) [inline test/e2e/autoscaling_utils.go]: Also I don't see any reason why this should be a separate function

File: test/e2e/autoscaling_utils.go [hunk 2]
```diff
@@ -204,7 +205,8 @@ func (rc *ResourceConsumer) CleanUp() {
 	expectNoError(rc.framework.Client.Experimental().HorizontalPodAutoscalers(rc.framework.Namespace.Name).Delete(rc.name, api.NewDeleteOptions(0)))
 }
 
-func runServiceAndRCForResourceConsumer(c *client.Client, ns, name string, replicas int) {
+// memLimit argument is in megabytes
+func runServiceAndRCForResourceConsumer(c *client.Client, ns, name string, replicas, cpuLimit, memLimit int) {
```
Comment (piosz) [inline test/e2e/autoscaling_utils.go]: Why cpuLimit, memLimit are not int64? In this case you will avoid conversion below

File: test/e2e/autoscaling_utils.go [hunk 3]
```diff
@@ -60,9 +60,10 @@ type ResourceConsumer struct {
 NewResourceConsumer creates new ResourceConsumer
 initCPU argument is in millicores
 initMemory argument is in megabytes
+memLimit argument is in megabytes
```
Comment (piosz) [inline test/e2e/autoscaling_utils.go]: Please add comment about cpuLimit as well. Please also add comment that both cpu and mem limit are the maximum amount of resources you expect to consume for a single pod. The situation when you specify replicas=1, initMemory=256M, memLimit=256M and then increase the memory consumption to 512M might be confusing without exhaustive explanation of what cpu/mem limit is.

File: test/e2e/autoscaling_utils.go [hunk 4]
```diff
@@ -204,7 +205,8 @@ func (rc *ResourceConsumer) CleanUp() {
 	expectNoError(rc.framework.Client.Experimental().HorizontalPodAutoscalers(rc.framework.Namespace.Name).Delete(rc.name, api.NewDeleteOptions(0)))
 }
 
-func runServiceAndRCForResourceConsumer(c *client.Client, ns, name string, replicas int) {
+// memLimit argument is in megabytes
```
Comment (piosz) [inline test/e2e/autoscaling_utils.go]: instead of writing this comment you can just rename the variable to sth like memLimitMb, cpuLimitMillis

**Comment by socaa** [comment]
PTAL

@piosz

**Comment by piosz** [comment]
LGTM

**Pull request merged**